iron-request.html

```diff
@@ -440,6 +440,23 @@
     _updateStatus: function() {
       this._setStatus(this.xhr.status);
       this._setStatusText((this.xhr.statusText === undefined) ? '' : this.xhr.statusText);
+    },
+
+    /**
+     * Returns the given response header in way that won’t cause an error during
+     * CORS requests if it is not returned or accessible due to
+     * `Access-Control-Expose-Headers`.
+     *
+     * @param {string} header The HTTP response header to return.
+     * @return {string|null} The header value or null if it is not present.
+     */
+    _responseHeader: function(header) {
+      var allHeaders = this.xhr.getAllResponseHeaders();
+      if (!!allHeaders &&
+          allHeaders.toLowerCase().indexOf(header.toLowerCase()) !== -1) {
```

Comment:
It is just causing noise in the console. Everything still seems to be
behaving correctly.

On Monday, November 9, 2015, Andrew Bonventre <notifications@github.com>
wrote:

> In iron-request.html
> <https://github.com/PolymerElements/iron-ajax/pull/147#discussion_r44354483>
> :
>
> > @@ -440,6 +440,23 @@
> >      _updateStatus: function() {
> >        this._setStatus(this.xhr.status);
> >        this._setStatusText((this.xhr.statusText === undefined) ? '' : this.xhr.statusText);
> > +    },
> > +
> > +    /**
> > +     * Returns the given response header in way that won’t cause an error during
> > +     * CORS requests if it is not returned or accessible due to
> > +     * `Access-Control-Expose-Headers`.
> > +     *
> > +     * @param {string} header The HTTP response header to return.
> > +     * @return {string|null} The header value or null if it is not present.
> > +     */
> > +    _responseHeader: function(header) {
> > +      var allHeaders = this.xhr.getAllResponseHeaders();
> > +      if (!!allHeaders &&
> > +          allHeaders.toLowerCase().indexOf(header.toLowerCase()) !== -1) {
>
> It's not throwing an exception so it can't be caught. It doesn't stop
> control flow in my tests.
> … <#150eec5c5e60cc79_>
> On Mon, Nov 9, 2015 at 6:31 PM Peter Burns <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> iron-request.html <#147 (comment)
> <https://github.com/PolymerElements/iron-ajax/pull/147#discussion_r44349819>>
> : > @@ -440,6 +440,23 @@ > _updateStatus: function() { >
> this._setStatus(this.xhr.status); >
> this._setStatusText((this.xhr.statusText === undefined) ? '' :
> this.xhr.statusText); > + }, > + > + /** > + * Returns the given response
> header in way that won’t cause an error during > + * CORS requests if it is
> not returned or accessible due to > + * `Access-Control-Expose-Headers`. >
> + * > + * @param <https://github.com/param> {string} header The HTTP
> response header to return. > + * @return <https://github.com/return>
> {string|null} The header value or null if it is not present. > + */ > +
> _responseHeader: function(header) { > + var allHeaders =
> this.xhr.getAllResponseHeaders(); > + if (!!allHeaders && > +
> allHeaders.toLowerCase().indexOf(header.toLowerCase()) !== -1) { @devinivy
> <https://github.com/devinivy> <https://github.com/devinivy> @hereiam133
> <https://github.com/hereiam133> <https://github.com/hereiam133> Is this
> causing requests to fail for you? We should add tests to cover your use
> cases if so. It's still a significant issue even if it's just putting a
> bunch of noise in the console, but it's much worse if it's causing requests
> to fail or block. — Reply to this email directly or view it on GitHub <
> https://github.com/PolymerElements/iron-ajax/pull/147/files#r44349819>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PolymerElements/iron-ajax/pull/147/files#r44354483>.
>